### ISSUE TITLE
Regression: Connection on OSX has broken since v1.3.0

### ISSUE THREAD

**Issue by SimonKagstrom** [adrienverge/openfortivpn#119]
With current git HEAD, I see these messages

```
INFO:   Setting new routes...
WARN:   Could not get current default route (No such file or directory).
INFO:   Adding VPN nameservers...
```

which seems to work with v1.3.0.

**Comment by DimitriPapadopoulos** [comment]
The warning is from this piece of code in **ipv4.c**:
```
	ret = ipv4_get_route(def_rt);
	if (ret != 0) {
		log_warn("Could not get current default route (%s).\n",
		         err_ipv4_str(ret));
		goto err_destroy;
	}
```
Function `ipv4_get_route()` does seem to have changed much since commit 5cb432beca51f777ff97b036d9352d46e08e488b “Mac OS X port”.

It might help - or not - if you could run `openfortivpn` with at least options `-v -v`.

**Comment by mrbaseman** [comment]
@mabin has a solution for this (see #99). However, we have to see how this fits into the current master branch.

**Comment by mrbaseman** [comment]
Sorry, I meant @mabin-J in my previous comment

**Comment by Mabin-J** [comment]
@DimitriPapadopoulos,
First of all, 'ipv4_get_route()' isn't changed since 5cb432b.

WARN message and broken 'split route mode' in MacOSX caused that #95, #97 is applied.
#95, #97 are for protecting route to VPN Server.
But caused 'ipv4_get_route()' cannot run correctly in MacOSX, 'split route mode' also broken.

**Comment by mrbaseman** [comment]
We have just committed #99 together with a few other improvements around this topic with https://github.com/adrienverge/openfortivpn/commit/d94915c02e18989021ce8aada5db78d98056192c 

In 1.3.0 we did not catch the case in which your VPN gateway pushes a route in which its own public IP is contained. We try to handle this now, but as @Mabin-J explains, the `ipv4_get_route()` needed for this has never been implemented for MacOSX. The warning you see just tells you that the code could not find out how your current default route is configured. We are working on a solution for this. 

Before https://github.com/adrienverge/openfortivpn/commit/d94915c02e18989021ce8aada5db78d98056192c not only a warning was issued but openfortivpn also refused to continue establishing the connection. With this commit at least 'split route mode' should be working again, although the above described protection for the route to the VPN gateway does not yet work on MacOSX. You can, however, manually add such a host route before starting openfortivpn as a workaround if needed.

'default route mode' (i.e. all traffic is routed through the VPN channel) does not yet work, because that mode obligatorily depends on a mechanism to protect routing towards the gateway. Rather than messing up the routing table completely we decided not to continue in this case.

To summarize: The warning is still there (and maybe even more detailed warnings now), but if it has worked for your setup up to 1.3.0 and it hast stopped working, it should work again now. We are aware that a few bits are still needed to cover all situations in MacOSX.

**Comment by SimonKagstrom** [comment]
OK, thanks! I'll try the fix soonish.

**Comment by SimonKagstrom** [comment]
Works with the latest git HEAD, thanks!

**Comment by mrbaseman** [comment]
I think that I have a solution for the above mentioned `ipv4_get_route()` function on MacOSX here: https://github.com/mrbaseman/openfortivpn/commit/07d866c3c934cc889563ad14e402a1cfa78fb609 Maybe you could test if this works for you? Thank you.

**Comment by Mabin-J** [comment]
@mrbaseman, Thank you for your comment.
I'll test when I arrive my office tomorrow.
